### PR TITLE
CI: scons fix abspath. Currently uncached for navd, locationd, opendbc

### PR DESCRIPTION
**Pull request by bongbui321**
**Description** [](A description of the bug and the fix. Also link any relevant issues.)
For #30161, translate most of absolute path to relative path. Currently uncached for navd, locationd, opendbc folder since I have to look more into why make relative path caching for these folders fail some CI tests. From the commai/openpilot master the cached scons build on docs CI test takes around 11s, after my relative path refactor is currently ~48s. 

**Verification** [](Explain how you tested this bug fix.)
image
image

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-  url = ../../commaai/opendbc.git
 [submodule "cereal"]
   path = cereal
   url = ../../commaai/cereal.git
 [submodule "rednose_repo"]
   path = rednose_repo
   url = ../../commaai/rednose.git
-[submodule "body"]
-  path = body
-  url = ../../commaai/body.git
```
Comment (adeebshihadeh) [inline .gitmodules:15]: can you open the PR for these submodules?

File: .github/workflows/docs.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ env:
 
   BUILD: selfdrive/test/docker_build.sh base
 
-  RUN: docker run --shm-size 1G -v $GITHUB_WORKSPACE:/tmp/openpilot -w /tmp/openpilot -e FILEREADER_CACHE=1 -e PYTHONPATH=/tmp/openpilot -e NUM_JOBS -e JOB_ID -e GITHUB_ACTION -e GITHUB_REF -e GITHUB_HEAD_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -v $GITHUB_WORKSPACE/.ci_cache/scons_cache:/tmp/scons_cache -v $GITHUB_WORKSPACE/.ci_cache/comma_download_cache:/tmp/comma_download_cache -v $GITHUB_WORKSPACE/.ci_cache/openpilot_cache:/tmp/openpilot_cache $BASE_IMAGE /bin/bash -c
+  RUN: docker run --shm-size 1G -v $GITHUB_WORKSPACE:/home/batman/openpilot -w /home/batman/openpilot -e FILEREADER_CACHE=1 -e PYTHONPATH=/home/batman/openpilot -e NUM_JOBS -e JOB_ID -e GITHUB_ACTION -e GITHUB_REF -e GITHUB_HEAD_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -v $GITHUB_WORKSPACE/.ci_cache/scons_cache:/tmp/scons_cache -v $GITHUB_WORKSPACE/.ci_cache/comma_download_cache:/tmp/comma_download_cache -v $GITHUB_WORKSPACE/.ci_cache/openpilot_cache:/tmp/openpilot_cache $BASE_IMAGE /bin/bash -c
```
Comment (adeebshihadeh) [inline .github/workflows/docs.yaml:18]: why change this path?

**Review by adeebshihadeh — COMMENTED** [review]
We'll also need a unit test for this

File: .github/workflows/docs.yaml (hunk 2 above)
Comment (bongbui321) [reply .github/workflows/docs.yaml:18]: I changed it for testing with relpath. Changing to `/home/batman` workspace ensures that CI doesn't cache for abspath, and the only way it is caching is through relpath.

**Pull request closed without merging**